blog/2018/12/15/practical-web-content-munging/index.markdown

```diff
@@ -0,0 +1,185 @@
+---
+status: published
+title: Practical Web Content Munging
+author: Joe Cooper
+images:
+  banner:
+    src: '/blog/2018/12/14/practical-web-content-munging/banner.jpg'
+    alt: 'An eyeball of alarming size'
+    data:
+      attribution: |-
+        Banner image: [An eyeball looking at Dallas](https://flic.kr/p/Rqj9Jj) by [Joe Cooper](https://www.flickr.com/photos/nerdnomad/) licensed [CC BY-NC-SA 2.0](https://creativecommons.org/licenses/by-nc-sa/2.0/)
+tags:
+  - advent
+  - css
+  - html
+data:
+  bio: swelljoe
+  description: Working with ugly old websites using Mojo::UserAgent and Mojo::DOM
+---
+Following brian d foy's great write-up of using Mojo::DOM selectors from Day 5, I thought it'd be fun to talk about some website migration scripts I recently built using Mojo::UserAgent and Mojo::DOM, two powerful aspects of Mojolicious. I've never used Mojo for a project before, but when I started looking at updating an old website that hasn't been updated in roughly 15 years (it's gotten new content, but the design is unchanged) Mojo::UserAgent and Mojo::DOM looked great. In the past I would have used regexes, and then I would likely have had to add a bunch of special cases where the regex didn't quite match, and then after a few hours of fighting I would have resigned myself to cleaning up the remaining handful of broken pages by hand. No more! Mojo::DOM always gets it right, and with alarmingly little code.
```

Comment:
Please link to the Day 5 article, and to modules the first time you reference them. For mojo modules please link to mojolicious.org/perldoc for others link to metacpan.